swe_roe/ex2.c

```diff
@@ -0,0 +1,1224 @@
+static char help[] = "Partial 2D dam break problem.\n";
+
+#include <assert.h>
+#include <math.h>
+#include <petsc.h>
+#include <petscdmda.h>
+#include <petscmat.h>
+#include <petscts.h>
+#include <petscvec.h>
+
+#define RDyAlloc(size, result) PetscCalloc1(size, result)
+
+typedef struct _n_User *User;
+
+typedef struct {
+  PetscReal X[3];
+} RDyCoordinate;
+
+typedef struct {
+  PetscReal V[3];
+} RDyVector;
+
+typedef enum {
+  CELL_TRI_TYPE = 0, /* tetrahedron cell for a 3D cell */
+  CELL_QUAD_TYPE     /* hexahedron cell for a 3D cell */
+} RDyCellType;
+
+typedef struct {
```

Comment:
Several of the comments for fields in this struct are copy-pasted and should probably be made more specific.